private/bufpkg/bufconfig/buf_yaml_file.go

```diff
@@ -306,20 +306,30 @@ func readBufYAMLFile(reader io.Reader, allowJSON bool) (BufYAMLFile, error) {
 		if err != nil {
 			return nil, err
 		}
+		// TODO: we do no validation of paths now
+		lintConfig, err := getLintConfigForExternalLint(
+			fileVersion,
+			externalBufYAMLFile.Lint,
+			normalpath.NormalizeAndValidate,
+		)
+		if err != nil {
+			return nil, err
+		}
+		// TODO: we do no validation of paths now
```

Comment:
This TODO is left as-is: in `bufcheck/internal`, `newConfigForRuleBuilders` does validation for these paths, which is also the behavior on main. Note that `newConfigForRuleBuilders` is called only for `buf lint` or `buf breaking`.

One interesting implication is that a user can have a semi-invalid buf.yaml (a file with the correct shape but invalid ignore paths) and they can successfully run `buf build` but not `buf lint`. I verified that this is the case with main. (The easiest way to try this out is to add `"."` to `ignore` and run `buf build` and `buf lint`)